DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_priority.py

```diff
@@ -0,0 +1,98 @@
+import asyncio
+
+import pytest
+from dent_os_testbed.utils.test_utils.tgen_utils import (tgen_utils_get_dent_devices_with_tgen,
+                                                         tgen_utils_setup_streams, tgen_utils_start_traffic,
+                                                         tgen_utils_stop_traffic, tgen_utils_get_traffic_stats,
+                                                         tgen_utils_get_loss, tgen_utils_dev_groups_from_config,
+                                                         tgen_utils_traffic_generator_connect,
+                                                         tgen_utils_get_egress_stats)
+from dent_os_testbed.utils.test_utils.br_utils import (configure_bridge_setup, configure_vlan_setup,
+                                                       get_traffic_port_vlan_mapping)
+
+pytestmark = [pytest.mark.suite_functional_vlan,
+              pytest.mark.asyncio,
+              pytest.mark.usefixtures("cleanup_bridges", "cleanup_tgen")]
+
+
+async def test_vlan_priority_tag(testbed):
+    """
+    Test Name: VLAN priority
+    Test Suite: suite_functional_vlan
+    Test Overview: Test packet forwarding with VLAN priority
+    Test Procedure:
+    1. Initiate test params.
+    2. Set links to vlans.
+    3. Map receiving and non receiving dut_ports.
+        Port 1 -> tx_port
+        Ports 2, 3, 4 -> rx_ports
+    4. Setup packet stream(s) for the broadcast packet.
+    5. Send traffic to rx_ports.
+    6. Verify traffic on receiving ports.
+    7. Verify priority was not stripped from the packet.
+    """
+
+    # 1. Initiate test params.
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 2)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgn connections")
+        return
+    device = dent_devices[0].host_name
+    tg_ports = tgen_dev.links_dict[device][0]
+    dut_ports = tgen_dev.links_dict[device][1]
+    vlan_priority = 6
+    vlan_id = 5
+
+    port_map = ({"port": 0, "settings": [{"vlan": vlan_id, "untagged": False, "pvid": True}]},
+                {"port": 1, "settings": [{"vlan": vlan_id, "untagged": False, "pvid": True}]})
+
+    await configure_bridge_setup(device, dut_ports)
+
+    # 2. Set links to vlans.
+    await configure_vlan_setup(device, port_map, dut_ports)
+
+    # 3. Map receiving and non receiving dut_ports.
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{"ixp": tg_ports[0], "ip": "100.1.1.2", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[1], "ip": "100.1.1.3", "gw": "100.1.1.6", "plen": 24, }])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, dut_ports, dev_groups)
+
+    tx_ports = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = [dev_groups[tg_ports[1]][0]["name"]]
+
+    # 4. Setup packet stream(s).
+    streams = {f"Traffic with VLAN priority": {
```

Comment:
How many packets and what rate does it send by default?

Reply:
The rate is `100`. 
How many packets i don't know. Do we need to send specific number of packets here?

Reply:
But we do check that there aren't any traffic loss

Reply:
Please send it on a line rate. At least 10k packets.